Celery / Background processing support for django-sendgrid-events

rorito opened the pull request:
Hi Patrick, we've been using django-sendgrid-events on https://www.tribute.co with good success. One thing we recently added in our fork was a quick check for some settings vars to enable background processing via celery workers. We use Heroku, so it frees up our web dynos for quicker request/response cycles and performs the work on a background worker. Anyways, here's a PR if you're interested : )

aspd172 committed:
#1006 - Support Celery Tasks

- updated to use process the sendgrid events on celery if the settings variable
  SENDGRID_BACKGROUND_PROCESSING is set to True

rorito committed:
Merge pull request #1 from aspd172/1006-support-celery-processing

#1006 - Support Celery Tasks

paltman commented:
I don't mind this at all, but don't want to require dependency on `celery`, which this change does.  I think it would be best to implement the `process_batch` method as a hookset ([example hookset](https://github.com/pinax/django-user-accounts/blob/master/account/hooks.py)) so you could override it at the project level to perform whatever queuing/backgrounding you wanted to do whether it was celery or some other background processor.

aspd172 committed:
#1211

- make sure the result is always json-serializable if the process is run
  in the background

rorito committed:
Merge pull request #2 from aspd172/1211-events-not-json-serializable

#1211 sendgrid Events not JSON serializable for celery

aspd172 committed:
#1211

- removed hard dependency on celery
- allowed the option to define a custom handler for sendgrid events

rorito committed:
Merge pull request #3 from aspd172/hookset-refactor

#1211

rorito committed:
Update README.rst

rorito committed:
Update README.rst

rorito commented:
@paltman Our developer Abel implemented the hookset refactor you were talking about and I added some instructions in the README about how to configure Celery support in your project. Thoughts?

emilioramirez committed:
added new column to show post_office.models.Email related object

rorito committed:
Merge pull request #4 from emilioramirez/master

added new column to show post_office.models.Email related object
